File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/timetable/Lesson.java
package com.happs.ximand.ringcontrol.model.object.timetable;

public class Lesson {

    private int number;
    private LessonScope lessonScope;

    public Lesson(int number) {
        this.number = number;
        this.lessonScope = new LessonScope(Time.getMidnightTime(), Time.getMidnightTime());
    }

    public Lesson(int number, Time startTime, Time endTime) {
        this.number = number;
        this.lessonScope = new LessonScope(startTime, endTime);
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public Time getStartTime() {
        return lessonScope.getStartTime();
    }

    public void setStartTime(Time startTime) {
        this.lessonScope.setStartTime(startTime);
    }

    public Time getEndTime() {
        return lessonScope.getEndTime();
    }

    public void setEndTime(Time endTime) {
        this.lessonScope.setEndTime(endTime);
    }

}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/timetable/Lesson.kt
package com.happs.ximand.ringcontrol.model.`object`.timetable

import com.happs.ximand.ringcontrol.model.`object`.timetable.Time.Companion.midnightTime

class Lesson {

    var number: Int
    private var lessonScope: LessonScope

    constructor(number: Int) {
        this.number = number
        lessonScope = LessonScope(midnightTime, midnightTime)
    }

    constructor(number: Int, startTime: Time, endTime: Time) {
        this.number = number
        lessonScope = LessonScope(startTime, endTime)
    }

    var startTime: Time
        get() = lessonScope.startTime
        set(startTime) {
            lessonScope.startTime = startTime
        }

    var endTime: Time
        get() = lessonScope.endTime
        set(endTime) {
            lessonScope.endTime = endTime
        }

}